TwitchPlaysAssembly/Src/ComponentSolvers/Modded/Misc/ResistorsComponentSolver.cs

```diff
@@ -48,20 +48,25 @@ public class ResistorsComponentSolver : ComponentSolver
 	
 	protected internal override IEnumerator RespondToCommandInternal(string inputCommand)
 	{
-		var commands = inputCommand.ToLowerInvariant().Trim().Split(new[] { ' ' }, StringSplitOptions.RemoveEmptyEntries);
+		string[] commands = inputCommand.ToLowerInvariant().Trim().Split(new[] { ' ' }, StringSplitOptions.RemoveEmptyEntries);
 
 		if (commands.Length >= 3 && commands.Length % 2 == 1 && commands[0].Equals("connect"))
 		{
-			IEnumerable<int?> pins = commands.Where((_, i) => i > 0).Select(pin => PinToIndex(pin));
+			IEnumerable<int?> pins = commands.Where((_, i) => i > 0).Select(PinToIndex);
 
-			if (pins.All(pinIndex => pinIndex != null))
+			IEnumerable<int?> pinIndices = pins as int?[] ?? pins.ToArray();
+			if (pinIndices.All(pinIndex => pinIndex != null))
 			{
 				yield return null;
 
-				foreach (int? pinIndex in pins)
+				foreach (int? pinIndex in pinIndices)
 				{
-					KMSelectable pinSelectable = _pins[(int) pinIndex];
-					DoInteractionClick(pinSelectable);
+					if (pinIndex != null)
```

Comment:
This check is useless, we already check to see if any pinIndexes are null on line 58.